src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -241,6 +246,8 @@ trait FastSync {
 
     private def handleActorTerminate(ref: ActorRef) = {
       context unwatch ref
+      val peer = assignedHandlers(ref)
+      peerRequestsTime -= peer
```

Comment:
Minor detail: As we remove the terminated peer from `peerRequestsTime`, should we do the same when handling `SyncRequestHandler.Done`?